ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+responsibility to delegate to a safe `IAccount` implementation.
+
+It is RECOMMENDED that each account leverages `IStandard` to validate and unpack operations, check **Reference
+Implementation** for examples. Account smart contract can be stateless to avoid sharing storage space with other delegated contracts.
+
+### Usage of Bytes
+
+There are many objects defined with `bytes` type for extensibility, future-compatibility purpose. All those objects are
+optional and their usages depends on `IStandard` and `IAccount` implementation.
+
+## Rationale
+
+> The main challenge with a generalized intent standard is being able to adapt to the evolving world of intents. Users
+> need to have a way to express their intents in a seamless way without having to make constant updates to their smart
+> contract wallets. --- ERC-7521
+
+The interface designs proposed in this EIP is inspired by ERC-7521, and the author tries to avoid turning EIP into a
+real implementation.
+
+### Execution in EOA contract code
+
+`SET_CODE_TX_TYPE=0x04` gives EOA the ability to execute code. Executing from the EOA brings some benefits:
+
+* Aligned with the nature of EOA that the execution is fully controlled by the account owner. EOA owner can easily turn
+  off all smart contract features by un-delegating the contract code.
+* `msg.sender` is always EOA address
+* Execution code can be stateless, this allows `account` to store no data.
+
+In the case that the EOA doesn't need to execute its contract code, or the intent cost is too high. The owner can use
+the account as EOA.
+
+### Validation in standard contract
+
+Validation logic commonly relies on contract state, for example, weighted multi-owner signature needs to track the
+weight of each signer. Keeping the functionality of intent validate purely inside `IStandard` makes it similar to
+the `EntryPoint` concept in ERC-4337, but a simpler version. Standard only taking responsibility of validation makes it
+easier for contract engineer to build, audit and maintain.
+
+Also, the `IStandard` interface can be considered as "modular". There can be a "compound" standard that
+breaks one intent into smaller pieces and call other standards to validate each piece and later combine the results
+together.
+
+### Gas abstraction
+
+This design allows any address to initiate transaction on behalf of the intent's sender, providing a gasless experience. The sender can define the "payment" to the sender in intent header or instruction, with transfers any token from the sender's account to `tx.origin`.
+
+### Auditability of both validation and execution
+
+It is very important both the standard and account implementation can be publicly audited and shared. The most important
+reason is security. And this can also help mediate the compatibility issues between standard and account.
```

Comment:
This is more content for the security considerations section.

Reply:
Updated 👍